cds-snc/forms-terraform
# Archive audit logs

# Summary | Résumé

closes https://github.com/cds-snc/forms-terraform/issues/337

- Implemented Audit Logs Archiver lambda ([Design Doc](https://docs.google.com/document/d/1-z87Dc9c75oYkdN0axDNzzJK8KGhdbRdXkVU8rRmwBQ))

# Test instructions

- Generate audit logs (manually using the Localstack UI https://app.localstack.cloud/inst/default/resources or through the use of the Forms application)
- Manually edit the `TimeStamp` of your audit logs using the https://app.localstack.cloud/inst/default/resources tool. Changed it so that the timestamp is at least 32 days old.
- Manually trigger the audit logs archiver lambda using `awslocal lambda invoke --function-name Audit_Logs_Archiver /dev/null`.
- The audit logs should now be archived. You can check both the DynamoDB and S3 services to see the result. The logs should have been removed from DynamoDB and saved in the `forms-${var.env}-audit-logs-archive-storage ` S3 bucket.

PS: Here is a sample of an audit log in case you want to manually add them in DynamoDB:
`{"Event#SubjectID#TimeStamp":"UserRegistration#clsngg0470002mgcphn6n2rqj#1708015693901","Status":"Archivable","Event":"UserRegistration","UserID":"clsngg0470002mgcphn6n2rqj","Subject":"User#clsngg0470002mgcphn6n2rqj","TimeStamp":1608015693996}`
